### PR TITLE
Add an update_user() method to UserClient

### PR DESCRIPTION
**Pull request by abretaud**
Hi!
I needed a way to update user emails using the api, so here it is in bioblend

### CODE REVIEW

File: bioblend/_tests/TestGalaxyUsers.py [hunk 1]
```diff
@@ -61,3 +61,27 @@ def test_get_current_user(self):
...
+        current_user = user['id']
+        user = self.gi.users.update_user(current_user, username=updated_username, email=updated_user_email)
+        user = self.gi.users.show_user(current_user)
+        self.assertEqual(user['username'], updated_username)
+        self.assertEqual(user['email'], updated_user_email)
+
+        if self.gi.config.get_config()['allow_user_deletion']:
+            deleted_user = self.gi.users.delete_user(user['id'])
+            self.assertEqual(deleted_user['email'], new_user_email)
+            self.assertTrue(deleted_user['deleted'])
```
Comment (nsoranzo) [inline bioblend/_tests/TestGalaxyUsers.py]: The 2 asserts are probably not needed.

File: bioblend/_tests/TestGalaxyUsers.py [hunk 2]
```diff
@@ -61,3 +61,27 @@ def test_get_current_user(self):
...
+        new_user_email = 'newuser2@example.com'
+        user = self.gi.users.create_local_user('newuser2', new_user_email, 'secret')
+        self.assertEqual(user['username'], 'newuser2')
+        self.assertEqual(user['email'], new_user_email)
+
+        updated_user_email = 'updateduser@example.com'
+        updated_username = 'updateduser'
+        current_user = user['id']
+        user = self.gi.users.update_user(current_user, username=updated_username, email=updated_user_email)
+        user = self.gi.users.show_user(current_user)
```
[the remark this answers is not in the document]
Comment (nsoranzo) [reply bioblend/_tests/TestGalaxyUsers.py]: Also can you just check the `user` dict returned by `self.gi.users.update_user()` ?

File: bioblend/galaxy/users/__init__.py [hunk 3]
```diff
@@ -186,3 +186,23 @@ def get_user_apikey(self, user_id):
...
+        :type username: str
+        :param username: Replace user name with the given string
+
+        :type email: str
+        :param email: Replace user email with the given string
+
+        :rtype: dict
+        :return: details of the updated user
+        """
+        url = self._make_url(user_id) + '/information/inputs'
```
Comment (nsoranzo) [inline bioblend/galaxy/users/__init__.py]: I see there is also a `PUT /api/users/{id}` endpoint (`update()` method in `lib/galaxy/webapps/galaxy/api/users.py`). I suppose this allows to change more attributes?

File: bioblend/_tests/TestGalaxyUsers.py (hunk 1 above)
Comment (nsoranzo) [reply bioblend/_tests/TestGalaxyUsers.py]: Any way the first assert is wrong, the 2nd parameter should be `updated_user_email`

File: bioblend/_tests/TestGalaxyUsers.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (abretaud) [reply bioblend/_tests/TestGalaxyUsers.py]: It's not returning the user info, just `{'message': 'User information has been saved.'}`

File: bioblend/_tests/TestGalaxyUsers.py [hunk 4]
```diff
@@ -61,3 +61,27 @@ def test_get_current_user(self):
...
+            self.skipTest('This Galaxy instance is not configured to use local users')
+        new_user_email = 'newuser2@example.com'
+        user = self.gi.users.create_local_user('newuser2', new_user_email, 'secret')
+        self.assertEqual(user['username'], 'newuser2')
+        self.assertEqual(user['email'], new_user_email)
+
+        updated_user_email = 'updateduser@example.com'
+        updated_username = 'updateduser'
+        user_id = user['id']
+        user = self.gi.users.update_user(user_id, username=updated_username, email=updated_user_email)
```
Comment (nsoranzo) [inline bioblend/_tests/TestGalaxyUsers.py]:
```suggestion
        self.gi.users.update_user(user_id, username=updated_username, email=updated_user_email)
```

File: bioblend/galaxy/users/__init__.py (hunk 3 above)
Comment (abretaud) [reply bioblend/galaxy/users/__init__.py]: I just tried with `PUT /api/users/{id}`, but no change is done in the db... Maybe the payload needs some special format?

Comment (nsoranzo) [reply bioblend/galaxy/users/__init__.py]: It seems only updating the username is supported in https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/managers/users.py#L586-L590

I think we should move the logic of user update from `UserAPIController.set_information()` to `UserDeserializer`, but for now this looks fine.

**Pull request merged**